src/main/scala/io/smartdatalake/workflow/action/Action.scala

```diff
@@ -72,6 +71,16 @@ private[smartdatalake] trait Action extends SdlConfigObject with ParsableFromCon
   def prepare(implicit session: SparkSession, context: ActionPipelineContext): Unit = {
     inputs.foreach(_.prepare)
     outputs.foreach(_.prepare)
+
+    // Make sure that data object names are still unique when replacing special characters with underscore
+    // Requirement from SQL transformations because temp view names can not contain special characters
+    val invalidTableNameCharacters = "[^a-zA-Z0-9_]".r
+    val duplicateNames = context.instanceRegistry.getDataObjects.map {
+      dataObj => invalidTableNameCharacters.replaceAllIn(dataObj.id.id, "_")
```

Comment:
How can we ensure that the replacement which is tested is the same which will be used?
https://github.com/smart-data-lake/smart-data-lake/blob/98bbcc3fa7337ad880f31340f52f4122ae05fea1/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomDfTransformer.scala#L76-L77
I would prefer to have the replacement of invalid table name characters to be implemented once only.

Reply:
Done in https://github.com/smart-data-lake/smart-data-lake/pull/41/commits/ebd8b215e2ec606ffd790f5f1b557f373b61d81f